### PR TITLE
Fixed the documentation for pandas.DataFrame.set_index inplace parameter

### PR DESCRIPTION
**Pull request by PurushothamanSrikanth**
Fixed the documentation for `pandas.DataFrame.set_index` 's `inplace` parameter.
Added proper explanation for True case.

### CODE REVIEW

**Review by arw2019 — APPROVED** [review]
LGTM modulo linting issue

File: pandas/core/frame.py [hunk 1]
```diff
@@ -4525,7 +4525,7 @@ def set_index(
         append : bool, default False
             Whether to append columns to existing index.
         inplace : bool, default False
-            Modify the DataFrame in place (do not create a new object).
+            If set to True, modifies the DataFrame in place (do not create a new object).
```
Comment (jbrockmendel) [inline pandas/core/frame.py:4528]: @PurushothamanSrikanth this line needs to be wrapped to make the linter pass (or remove "set to" might do it)

**Comment by PurushothamanSrikanth** [comment]
@arw2019 @pep8speaks Is it fine now?

**Comment by jbrockmendel** [comment]
thanks @PurushothamanSrikanth

**Pull request merged**